### PR TITLE
chore(events): refactor event structure

### LINKED ISSUE #4112: Refactor Event Structure for Improved Clarity and Consistency
Proposed Changes:

Field Renaming:

- Rename `parent` to `ancestors` to more accurately reflect its content (array of ancestor processes).
- Rename `pid` (under `process`) to `hostPid` to distinguish it from the namespaced PID.
- Rename `namespacedPid` to `pid` (under `process`) for consistency with standard terminology.
- Rename `tid` (under thread) to `hostTid`
- Rename `namespacedTid` to `tid` (under `thread`) 
- Rename `started` (under `container`) to `isRunning` to clarify its meaning.
- Rename `start` (under `thread`) to `startTime` for consistency with other timestamp fields.
- Rename `entityId` to `uniqueId` to be more descriptive and align with existing documentation.

Structural Change:
Change ancestors from a nested object to an array of objects to simplify representation and analysis of process lineage.

### PR DESCRIPTION
**Pull request by rscampos**
### 1. Explain what the PR does

fix #4112

3293ddbb0 **grpc: refactor event structure**
b0f99a53c **api: event definitions api**

3293ddbb0 **grpc: refactor event structure**

```
- rename some protobuf fields;
- ancestors (deprecated parent field) is now array of objects.
```

### CODE REVIEW

File: pkg/server/grpc/tracee.go [hunk 1]
```diff
@@ -779,27 +779,29 @@ func getProcess(e trace.Event) *pb.Process {
 	}
 
 	return &pb.Process{
-		Executable:    executable,
-		EntityId:      wrapperspb.UInt32(e.ProcessEntityId),
-		Pid:           wrapperspb.UInt32(uint32(e.HostProcessID)),
-		NamespacedPid: wrapperspb.UInt32(uint32(e.ProcessID)),
+		Executable: executable,
+		UniqueId:   wrapperspb.UInt32(e.ProcessEntityId),
+		Pid:        wrapperspb.UInt32(uint32(e.HostProcessID)),
```
Comment (yanivagman) [inline pkg/server/grpc/tracee.go:784]: This should be HostPid since we initialize it with e.HostProcrssID

File: pkg/server/grpc/tracee.go [hunk 2]
```diff
@@ -779,27 +779,29 @@ func getProcess(e trace.Event) *pb.Process {
...
+		HostPid:    wrapperspb.UInt32(uint32(e.ProcessID)),
```
Comment (yanivagman) [inline pkg/server/grpc/tracee.go:785]: This should be Pid

File: pkg/server/grpc/tracee.go [hunk 3]
```diff
@@ -779,27 +779,29 @@ func getProcess(e trace.Event) *pb.Process {
...
 		RealUser: &pb.User{
 			Id: wrapperspb.UInt32(uint32(e.UserID)),
 		},
 		Thread: &pb.Thread{
-			Start:          threadStartTime,
+			StartTime:      threadStartTime,
 			Name:           e.ProcessName,
-			EntityId:       wrapperspb.UInt32(e.ThreadEntityId),
+			UniqueId:       wrapperspb.UInt32(e.ThreadEntityId),
 			Tid:            wrapperspb.UInt32(uint32(e.HostThreadID)),
```
Comment (yanivagman) [inline pkg/server/grpc/tracee.go:793]: This should be HostTid

File: pkg/server/grpc/tracee.go [hunk 4]
```diff
@@ -779,27 +779,29 @@ func getProcess(e trace.Event) *pb.Process {
...
-			NamespacedTid:  wrapperspb.UInt32(uint32(e.ThreadID)),
+			HostTid:        wrapperspb.UInt32(uint32(e.ThreadID)),
```
Comment (yanivagman) [inline pkg/server/grpc/tracee.go:794]: This should be Tid

File: pkg/server/grpc/tracee_test.go [hunk 5]
```diff
@@ -40,17 +40,17 @@ func Test_convertEventWithProcessContext(t *testing.T) {
 	protoEvent, err := convertTraceeEventToProto(traceEvent)
 	assert.NoError(t, err)
 
-	assert.Equal(t, uint32(1), protoEvent.Context.Process.NamespacedPid.Value)
-	assert.Equal(t, uint32(2), protoEvent.Context.Process.Thread.NamespacedTid.Value)
+	assert.Equal(t, uint32(1), protoEvent.Context.Process.HostPid.Value)
```
Comment (yanivagman) [inline pkg/server/grpc/tracee_test.go:43]: This should be Pid

File: pkg/server/grpc/tracee_test.go [hunk 6]
```diff
@@ -40,17 +40,17 @@ func Test_convertEventWithProcessContext(t *testing.T) {
...
+	assert.Equal(t, uint32(2), protoEvent.Context.Process.Thread.HostTid.Value)
```
Comment (yanivagman) [inline pkg/server/grpc/tracee_test.go:44]: This should be Tid

**Review by yanivagman — CHANGES_REQUESTED** [review]

**Review by yanivagman — APPROVED** [review]
LGTM

**Comment by rscampos** [comment]
@yanivagman 
API: https://github.com/aquasecurity/tracee/pull/4115

**Pull request closed without merging**